app/lib/controllers/chat_room_controller.dart

```diff
@@ -341,24 +387,12 @@ class ChatRoomController extends GetxController {
     } else if (msgtype == 'm.key.verification.request') {}
   }
 
-  int _findMessage(String eventId) {
-    for (var i = 0; i < messages.length; i++) {
-      if (messages[i].id == eventId) {
-        return i;
-      }
-    }
-    return -1;
-  }
-
   void sendButtonUpdate() {
     isSendButtonVisible = textEditingController.text.trim().isNotEmpty;
     update();
   }
 
-  @override
-  void onClose() {
-    super.onClose();
-    textEditingController.dispose();
-    focusNode.removeListener(() {});
+  Future<bool> typingNotice(bool typing) async {
```

Comment:
Here too.

Reply:
`typingNotice()` is for sending notification about my typing.
We have to implement sending of typing event also, so that peer can know if I am typing.